### PR TITLE
feat(oracle): support installing RHCK kernel 

### PR DESCRIPTION
**Pull request by jkoelker**
**What problem does this PR solve?**:

Adds the `oracle_kernel` varaible for ansible. When set to the string `RHCK` it will install the RHCK kernel and set it as the default for the next boot.

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-88561

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by jkoelker** [comment]
Just a rebase on `main` to keep the CI #fresh

**Comment by jkoelker** [comment]
The CI failures are all the expected ones. The OVA offlines are timing out shutting down, and the CentOS 7 and Azure Ubuntu Nvidia are still WIP.

**Review by alejandroEsc — APPROVED** [review]
I have limited ability to understand some of this but it looks like it doing things correctly, I also see that there is some refactoring.

File: ansible/roles/packages/tasks/oracle.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+
```
Comment (msdolbey) [inline ansible/roles/packages/tasks/oracle.yaml:2]: I really don't know all the yamls in the world, and this probably doesn't really matter at all - but should there be this extra line here?

**Comment by jkoelker** [comment]
> Do we need an E2E variant for this?

I would like one, but it requires running the image to actually check.

I attempted to add a goss step to check the grub entry, but it was turning into more hassle to shovel the flag around and then goss only checked the running kernel, not the configured grub entry.

I suppose we could add a build to make sure that the ansible doesn't fail, but it's only 1/2 the story since it may succeed in installing it and adding it, but the grubenv / defaults gets set later on. I guess its better than nothing. I'll add that this weekend.

**Pull request merged**